commit.go

```diff
@@ -177,7 +186,8 @@ func (c *Commit) Encode(to io.Writer) (n int, err error) {
 	n = n + n2
 
 	for _, hdr := range c.ExtraHeaders {
-		n3, err := fmt.Fprintf(to, "%s %s\n", hdr.K, hdr.V)
+		n3, err := fmt.Fprintf(to, "%s %s\n",
+			hdr.K, strings.Replace(hdr.V, "\n", "\n ", -1))
```

Comment:
In future versions of Go, this line can be replaced with `strings.ReplaceAll(hdr.V, "\n", "\n ")`, but this was introduced in Go 1.12, which we don't build against yet.